tests/triggerWorkflow.test.ts

```diff
@@ -310,19 +247,63 @@ describe("triggerWorkflow Tests", () => {
       isBlocking: true,
     });
 
-    console.log("triggerWorkflow result", result);
-
     // The list should now contain one execution
-    const executions2 = await client.getExecutions(workflowId);
-    console.log("executions2", executions2);
+    const executions2 = await client.getExecutions([workflowId]);
     expect(Array.isArray(executions2.result)).toBe(true);
     expect(executions2.result.length).toEqual(1);
 
     const workflow = await client.getWorkflow(workflowId);
     expect(workflow.status).toEqual(WorkflowStatuses.COMPLETED);
     expect(workflow.totalExecution).toEqual(1);
+    expect(executions2.result[0].triggerMetadata.blockNumber).toEqual(blockNumber+5);
+    expect(executions2.result[0].triggerMetadata.txHash).toEqual("0x1234567890");
+    expect(executions2.result[0].triggerMetadata.type).toEqual(TriggerTypes.EVENT);
   });
 
+  test("trigger return correct execution id in blocking mode", async () => {
+    const wallet = await client.getWallet({ salt: "0" });
+    await cleanupWorkflows(client, wallet.address);
+    const epoch = Math.floor(Date.now() / 1000);
+
+    // Create a cron trigger with a schedule of every minute
+    const trigger = TriggerFactory.create({
+      name: "cronTrigger",
+      type: TriggerTypes.CRON,
+      data: { scheduleList: ["* * * * *"] },
+    });
+
+    const workflowId = await client.submitWorkflow(
+      client.createWorkflow({
+        ...WorkflowTemplate,
+        trigger,
+        smartWalletAddress: wallet.address,
+      })
+    );
+    queueForRemoval(createdWorkflows, workflowId);
+
+    const executions = await client.getExecutions([workflowId]);
+
+    // The list should be empty because the workflow has not been executed yet
+    expect(Array.isArray(executions.result)).toBe(true);
+    expect(executions.result.length).toEqual(0);
+
+    const result = await client.triggerWorkflow({
+      id: workflowId,
+      data: {
+        type: TriggerTypes.CRON,
+        epoch: epoch + 60, // set epoch to 1 minute later
+      },
+      isBlocking: true,
+    });
+
+    // The list should now contain one execution, the id from manual trigger should matched
+    const execution = await client.getExecution(workflowId, result.executionId);
```

Comment:
Make sense, but I’ve got a question.

The isBlocking is to wait for the exeuction to complete, right? My understanding is that:

When `isBlocking: true`, the `triggerWorkflow` waits for execution to complete, at this point, it can get the result of getExecution.

Otherwise, when `isBlocking: false`, the `triggerWorkflow` runs in async mode, that it will return a tracking id immediately, which can be used later to `getExecution`. Also, we might need to define a response for `pending`, in case the getExecution is called too quickly.

Let me know if the two cases make sense. From the Studio, we would always run it in isBlocking mode, and wait for the executiond details.